elasticsearch.go

```diff
@@ -297,9 +300,16 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 
 	// ResponseCheck path continues, we run the header check on the first answer from ES.
 	if err == nil {
-		checkHeader := func() error { return genuineCheckHeader(res.Header) }
+		checkHeader := func() error {
+			if res != nil {
+				return genuineCheckHeader(res.Header)
+			}
+			return nil
+		}
```

Comment:
As we have already checked `err == nil` above, res must be non-nil according to [net/http](https://pkg.go.dev/net/http#Client.Do):

> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close. If the Body is not both read to EOF and closed, the Client's underlying RoundTripper (typically Transport) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

Reply:
Nice catch, thanks for this! 👍